From 5cbe4522fce363fef1e338ddd201c35a49231628 Mon Sep 17 00:00:00 2001
From: Shaohua Li <shli@kernel.org>
Date: Wed, 19 Feb 2014 20:20:21 +0800
Subject: [PATCH] blk-mq: add REQ_SYNC early
Git-commit: 739c3eea711a255df5ed1246face0a4dce5e589f
Patch-mainline: v3.14-rc6
References: fate#315209

Add REQ_SYNC early, so rq_dispatched[] in blk_mq_rq_ctx_init
is set correctly.

Signed-off-by: Shaohua Li<shli@fusionio.com>
Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jan Kara <jack@suse.cz>

---
 block/blk-mq.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 1b8b50d..883f720 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -860,6 +860,8 @@ static void blk_mq_make_request(struct request_queue *q, struct bio *bio)
 	ctx = blk_mq_get_ctx(q);
 	hctx = q->mq_ops->map_queue(q, ctx->cpu);
 
+	if (is_sync)
+		rw |= REQ_SYNC;
 	trace_block_getrq(q, bio, rw);
 	rq = __blk_mq_alloc_request(hctx, GFP_ATOMIC, false);
 	if (likely(rq))
-- 
1.6.0.2

